homeassistant/components/sia/alarm_control_panel.py

```diff
@@ -0,0 +1,292 @@
+            and state.state is not None
+            and state.state
+            in [
+                STATE_ALARM_ARMED_AWAY,
+                STATE_ALARM_ARMED_CUSTOM_BYPASS,
+                STATE_ALARM_ARMED_NIGHT,
+                STATE_ALARM_DISARMED,
+                STATE_ALARM_TRIGGERED,
+                STATE_UNKNOWN,
+            ]
+        ):
+            self.state = state.state
+        else:
+            self.state = None
+        self._async_track_unavailable()
+        async_dispatcher_connect(
+            self.hass, DATA_UPDATED, self._schedule_immediate_update
+        )
+        self._unsub = self.hass.bus.async_listen(
+            self._event_listener_str, self.async_handle_event
+        )
+        self.async_on_remove(self._sia_on_remove)
+
+    @callback
+    def _sia_on_remove(self):
+        """Remove the unavailability and event listener."""
+        if self._unsub:
+            self._unsub()
+        if self._remove_unavailability_tracker:
+            self._remove_unavailability_tracker()
+
+    @callback
+    def async_handle_event(self, event: Event):
+        """Listen to events for this port and account and update states.
+
+        If the port and account combo receives any message it means it is online and can therefore be set to available.
+        """
+        self.assume_available()
+        new_state = CODE_CONSEQUENCES.get(event.data[EVENT_CODE])
+        if (
+            int(event.data[EVENT_ZONE]) != self._zone
+            or self.registry_entry.disabled
+            or new_state is None
+        ):
+            return
+        self._attr.update(
+            {
+                ATTR_LAST_MESSAGE: event.data[EVENT_MESSAGE],
+                ATTR_LAST_CODE: event.data[EVENT_CODE],
+                ATTR_LAST_TIMESTAMP: event.data[EVENT_TIMESTAMP],
+            }
+        )
+        self.state = new_state
+        self.async_write_ha_state()
+
+    @callback
+    def _schedule_immediate_update(self):
+        self.async_write_ha_state()
+
+    @property
+    def name(self) -> str:
+        """Get Name."""
+        return self._name
+
+    @property
+    def ping_interval(self) -> int:
+        """Get ping_interval."""
+        return str(self._ping_interval)
+
+    @property
+    def state(self) -> str:
+        """Get state."""
+        return self._state
+
+    @property
+    def account(self) -> str:
+        """Return device account."""
+        return self._account
+
+    @property
+    def unique_id(self) -> str:
+        """Get unique_id."""
+        return self._unique_id
+
+    @property
+    def available(self) -> bool:
+        """Get availability."""
+        return self._is_available
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Return device attributes."""
+        return self._attr
+
+    @property
+    def should_poll(self) -> bool:
+        """Return False if entity pushes its state to HA."""
+        return False
+
+    @state.setter
```

Comment:
```suggestion
    @property
```

Reply:
We should not set the state from outside the entity directly.  If you need to change the state from the outside, use the dispatcher